### PR TITLE
fix bugs in healthmgr and metricscachemgr

### PR DESCRIPTION
**Pull request by huijunw**
1. Fix bugs in the healthmgr and metricscahemgr
2. Add logs
3. Update metricscache query

### CODE REVIEW

File: heron/healthmgr/src/java/com/twitter/heron/healthmgr/diagnosers/SlowInstanceDiagnoser.java [hunk 1]
```diff
@@ -76,6 +80,7 @@ public Diagnosis diagnose(List<Symptom> symptoms) {
       }
     }
 
+    LOG.info("resultSymptom " + resultSymptom);
```
Comment (objmagic) [inline heron/healthmgr/src/java/com/twitter/heron/healthmgr/diagnosers/SlowInstanceDiagnoser.java]: `result symptom:`

Comment (huijunw) [reply heron/healthmgr/src/java/com/twitter/heron/healthmgr/diagnosers/SlowInstanceDiagnoser.java]: done

File: heron/healthmgr/src/java/com/twitter/heron/healthmgr/diagnosers/SlowInstanceDiagnoser.java [hunk 2]
```diff
@@ -45,6 +45,9 @@ public Diagnosis diagnose(List<Symptom> symptoms) {
         || !processingRateSkewComponents.isEmpty()) {
       // Since there is no back pressure or disparate wait count or similar
       // execution count, no action is needed
+      LOG.info("no back pressure " + bpSymptoms.isEmpty());
+      LOG.info("no disparate wait count " + waitQDisparityComponents.isEmpty());
+      LOG.info("no similar execution count " + !processingRateSkewComponents.isEmpty());
```
Comment (ashvina) [inline heron/healthmgr/src/java/com/twitter/heron/healthmgr/diagnosers/SlowInstanceDiagnoser.java]: These logs seem noisy? Diagnosers will be invoked periodically/frequently and I am not sure how these logs will help the user debug an issue.

File: heron/healthmgr/src/java/com/twitter/heron/healthmgr/diagnosers/SlowInstanceDiagnoser.java [hunk 3]
```diff
@@ -56,6 +59,7 @@ public Diagnosis diagnose(List<Symptom> symptoms) {
     ComponentMetrics pendingBufferMetrics = waitQDisparityComponents.get(bpMetrics.getName());
     if (pendingBufferMetrics == null) {
       // no wait Q disparity for the component with back pressure. There is no slow instance
+      LOG.info("no wait Q disparity for the component with back pressure.");
```
Comment (ashvina) [inline heron/healthmgr/src/java/com/twitter/heron/healthmgr/diagnosers/SlowInstanceDiagnoser.java]: This log message does not look very helpful. Either remove it or augment with wait q sizes and back pressure metrics.

File: heron/healthmgr/src/java/com/twitter/heron/healthmgr/diagnosers/SlowInstanceDiagnoser.java [hunk 4]
```diff
@@ -76,6 +80,7 @@ public Diagnosis diagnose(List<Symptom> symptoms) {
       }
     }
 
+    LOG.info("result symptom: " + resultSymptom);
```
Comment (ashvina) [inline heron/healthmgr/src/java/com/twitter/heron/healthmgr/diagnosers/SlowInstanceDiagnoser.java]: Seems noisy and not very helpful during normal execution or when the problem is in other components.

File: heron/healthmgr/src/java/com/twitter/heron/healthmgr/diagnosers/SlowInstanceDiagnoser.java (hunk 2 above)
Comment (huijunw) [reply heron/healthmgr/src/java/com/twitter/heron/healthmgr/diagnosers/SlowInstanceDiagnoser.java]: what we observed is the topology is in backpressure state, but the detector failed to catch
```
[2017-10-03 21:00:26 +0000] [WARNING] com.twitter.heron.healthmgr.policy.AutoRestartBackpressureContainerPolicy: Unknown diagnoses. None resolver selected. 
```
I think these logs helps to find out why the detector failed to detect backpressure

File: heron/healthmgr/src/java/com/twitter/heron/healthmgr/diagnosers/SlowInstanceDiagnoser.java (hunk 3 above)
Comment (huijunw) [reply heron/healthmgr/src/java/com/twitter/heron/healthmgr/diagnosers/SlowInstanceDiagnoser.java]: updated log message

File: heron/healthmgr/src/java/com/twitter/heron/healthmgr/diagnosers/SlowInstanceDiagnoser.java (hunk 4 above)
Comment (huijunw) [reply heron/healthmgr/src/java/com/twitter/heron/healthmgr/diagnosers/SlowInstanceDiagnoser.java]: we found this function returned null. i think this log may help to find out why the detector failed to catch the abnormals.

**Comment by huijunw** [comment]
@ashvina i have updated the pr according to your comments.
Do not worry about the metricscache part. Mark already reviewed that part.
Once you find comfortable with dhalion part, we can merge.

File: heron/metricscachemgr/src/java/com/twitter/heron/metricscachemgr/MetricsCacheManagerHttpServer.java [hunk 5]
```diff
@@ -95,28 +92,12 @@ public static void main(String[] args)
...
-      System.out.println("MetricsCacheMgr is not ready");
-      return;
+          "Usage: java MetricsQuery <host:port> <component_name> <metrics_name>");
     }
 
     // construct metric cache stat url
-    String url = "http://" + location.getHost() + ":" + location.getStatsPort()
-        + MetricsCacheManagerHttpServer.PATH_STATS;
+    String url = "http://" + args[0] + MetricsCacheManagerHttpServer.PATH_STATS;
+    System.out.println("endpoint: " + url + "; component: " + args[1]);
```
Comment (nlu90) [inline heron/metricscachemgr/src/java/com/twitter/heron/metricscachemgr/MetricsCacheManagerHttpServer.java]: is this `println` for debugging purpose？

Comment (huijunw) [reply heron/metricscachemgr/src/java/com/twitter/heron/metricscachemgr/MetricsCacheManagerHttpServer.java]: this is a tool.
it echo prints user input params.

**Comment by huijunw** [comment]
This PR will be separated into small PRs.

**Comment by huijunw** [comment]
This PR is replaced by #2401 and #2406, so close it.

**Pull request closed without merging**